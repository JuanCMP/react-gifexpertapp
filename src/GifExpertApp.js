
import React, { useState } from 'react';

// components
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = ( {defaultCategories = []} )  => {
    
    // Status hook
    const [categories, setCategories ] = useState( defaultCategories );
    
    // JSX
    return (
        <>
            <h2>You can find GIF's that you want!</h2>
            
            {/* 'AddCategory' and 'GifExpertApp' communication */}
            <AddCategory setCategories = { setCategories }/>
            <hr />
            <ol>
                {
                    categories.map( category => (
                        <GifGrid
                            key = {category}
                            category = {category}  
                        />
                    ))
                }
            </ol>
            <p>Developed by Juan C. Murcia.</p>
        </>
    )
}







