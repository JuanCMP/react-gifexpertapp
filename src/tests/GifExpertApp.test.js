import { shallow } from 'enzyme';
import React from 'react';
import { GifExpertApp } from '../GifExpertApp';

describe( 'tests GifExpertApp', () => {
    
    test( 'make GifExpertApp compatible', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    
    test( 'Matched categories list', () => {
        
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GifExpertApp defaultCategories={categories}/> );

        expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find( 'GifGrid' ).length).toBe( categories.length );
    } )
})

