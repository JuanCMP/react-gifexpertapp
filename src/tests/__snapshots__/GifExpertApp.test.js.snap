// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests GifExpertApp Match categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`tests GifExpertApp Matched categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`tests GifExpertApp Show categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`tests GifExpertApp make GifExpertApp compatible 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
